### ISSUE TITLE
Sufficient equipment required to implement

### ISSUE THREAD

**Issue by okina13070358** [VinAIResearch/ISBNet#24]
Hi, thank you for your good model.
Now, I'm trying implemantation for my dataset. So these days, I practice sample datasets (e.g. S3DIS, STLPS3D).  I've trained backbone,  but I could not train entire model by cuda memory error.
So I tried these approach:
1. convert from fp16=False to fp16=True
2. reduced epoch, batch and crop size

But cuda memory error happend every approach.
I guess my equipments are lack of specs. What is the minimum GPU memory required to use full-spec ISBNet?

P.S
I'm using geforce 3060 ti (8GB memory). Are my equipment not enough?

Regards.

**Comment by jongmin4422** [comment]
I have tried similar things, but I have encountered a cuda out of memory error.

Actually, I am not the author of this paper, but in this paper, author says that they train a model using one v100 gpu.
The memory size of the gpu I have is 24gb, but seeing that I encountered the cuda out of memory error, I think 8gb of memory may not be enough to learn.

I hope it'll be helpful for your research.

**Comment by okina13070358** [comment]
Hi jongmin4422, thank you for your reply. I'm surprised cuda memory error happens with 24GB. I agree with your opinion.

I understand what state I'm in. But I would like to hear the comments about this from the author. So I'll leave open for a while (one or two week).

**Comment by ngoductuanlhp** [comment]
We conducted our experiments on a V100 GPU with 32GB memory and occasionally encountered out-of-memory (OOM) issues when running datasets with a large number of points, such as S3DIS. Unfortunately, I regret that a GPU with 8GB memory is insufficient to train our ISBNet, even with adjustments like using FP16 or reducing batch size and number of points.

When testing with a 24GB memory GPU and facing OOM, I guess the bottleneck in the mask_heads_forward() function, specifically in the dot product operation between numerous queries and thousands of points. To address this, we devised a solution outlined in our repository at https://github.com/VinAIResearch/ISBNet/blob/ae02fcb3db609600b7576f2d4fec91ead5773664/isbnet/model/isbnet.py#L762. This trick involves splitting queries into multiple chunks and performing convolution chunk by chunk to circumvent the OOM problem. I hope this could resolve your problem.
